### ISSUE TITLE
Enable enhanced text selection by default

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#7584]
With #7539 merged, we should address the following items before we can enable the feature by default:
- [ ] Use a red-black tree or binary search in `expandBoundsLTR` (see inline TODO item)
- [ ] Use reference counting in `expandBoundsLTR` (see inline TODO item)
- [x] Remove `dataset` usage in `text_layer.js` as there is unnecessary type conversion
- [x] Improve handling cases where each glyph is in a separate div (see https://github.com/mozilla/pdf.js/pull/7539#issuecomment-243205897)
- [x] Make enhance text selection a viewer preference

**Comment by Snuffleupagus** [comment]
I'm proposing that we close this as WONTFIX for the time being, since:
 - No real work, to address the remaining points, have happened for a couple years and the `enhanceTextSelection`-mode is essentially not being worked on or improved at all.

 - Even with the aforementioned improvements implemented (which only affects the initial parsing), the `enhanceTextSelection`-mode still won't perform well for pages that contain *a lot* of textLayer-elements during actual user-interaction.
   The explanation here is the need to update the DOM as the user clicks/releases the mouse button, for potentially thousands of elements, which unfortunately will feel sluggish and result in a bad UX in larger and/or complex PDF documents.
